### PR TITLE
Add StudentList Package and Class

### LINKED ISSUE #25: Student List Package
Add StudentList class to store inputs.

### PR DESCRIPTION
**Pull request by ehz0ah**
- Add StudentList Package and Class
- Updated Student Class:
  1. Remove name
  2. Change id from int to String
  3. Change UniPreferences from static array to ArrayList
-  Updated Parser class to be more comprehensive by combining with the StudentList class
- Tested new IO redirection test (AddCommand, DeleteCommand & ExitCommand)
- Refactor Duke Class to FindOurSEP 

Fix #25

### CODE REVIEW

File: src/main/java/student/Student.java [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
+
 public class Student {
-    private String name;
-    private int id;
+    private String id;
     private float gpa;
-    private int[] uniPreferences = new int[3];
+    private ArrayList<Integer> uniPreferences;
     private boolean isSuccessfullyAllocated = false;
+    private int allocatedUniversity = -1;
```
Comment (paulktham) [inline src/main/java/student/Student.java:10]: do you think we should set allocatedUniversity to be a university type? In the future will it be more convenient for our code?

File: src/main/java/student/Student.java [hunk 2]
```diff
@@ -1,32 +1,29 @@
...
 
+import java.util.ArrayList;
+
 public class Student {
-    private String name;
-    private int id;
+    private String id;
     private float gpa;
-    private int[] uniPreferences = new int[3];
+    private ArrayList<Integer> uniPreferences;
```
Comment (paulktham) [inline src/main/java/student/Student.java:8]: this is good, makes the code more flexible

**Review by isaacsaw25 — APPROVED** [review]

File: src/main/java/student/Student.java (hunk 1 above)
Comment (ehz0ah) [reply src/main/java/student/Student.java:10]: Do you guys mean an enum?

**Review by isaacsaw25 — APPROVED** [review]
LGTM!

**Pull request merged**